browser/brave_wallet/brave_wallet_service_delegate_impl_android.cc

```diff
@@ -0,0 +1,55 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_wallet/brave_wallet_service_delegate_impl_android.h"
+
+#include <utility>
+
+#include "brave/components/permissions/contexts/brave_ethereum_permission_context.h"
+#include "chrome/browser/profiles/profile.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/web_contents.h"
+#include "url/gurl.h"
+
+namespace brave_wallet {
+
+BraveWalletServiceDelegateImpl::BraveWalletServiceDelegateImpl(
+    content::BrowserContext* context)
+    : context_(context), weak_ptr_factory_(this) {}
+
+BraveWalletServiceDelegateImpl::~BraveWalletServiceDelegateImpl() = default;
+
+void BraveWalletServiceDelegateImpl::AddEthereumPermission(
+    const std::string& origin_spec,
+    const std::string& account,
+    AddEthereumPermissionCallback callback) {
+  bool success =
+      permissions::BraveEthereumPermissionContext::AddEthereumPermission(
+          context_, origin_spec, account);
+  std::move(callback).Run(success);
+}
+
+void BraveWalletServiceDelegateImpl::HasEthereumPermission(
+    const std::string& origin_spec,
+    const std::string& account,
+    HasEthereumPermissionCallback callback) {
+  bool has_permission = false;
+  bool success =
+      permissions::BraveEthereumPermissionContext::HasEthereumPermission(
+          context_, origin_spec, account, &has_permission);
+  std::move(callback).Run(success, has_permission);
+}
+
+void BraveWalletServiceDelegateImpl::ResetEthereumPermission(
+    const std::string& origin_spec,
+    const std::string& account,
+    ResetEthereumPermissionCallback callback) {
+  bool success =
+      permissions::BraveEthereumPermissionContext::ResetEthereumPermission(
+          context_, origin_spec, account);
+  std::move(callback).Run(success);
+}
```

Comment:
Seems slightly unideal to me to have these duplicated, these are exactly the same as desktop implementation.
Would be great if we can avoid this, but I can't think of a better alternative for now.
What this PR currently does seem to be the easiest to read and at least the functions here are pretty trivial.